internal/controller/controllers/infraenv_controller.go

```diff
@@ -176,6 +172,9 @@ func (r *InfraEnvReconciler) updateInfraEnv(ctx context.Context, log logrus.Fiel
 	if len(infraEnv.Spec.AdditionalNTPSources) > 0 {
 		updateParams.InfraEnvUpdateParams.AdditionalNtpSources = swag.String(strings.Join(infraEnv.Spec.AdditionalNTPSources[:], ","))
 	}
+	if infraEnv.Spec.ImageType != common.ImageTypeValue(internalInfraEnv.Type) {
+		updateParams.InfraEnvUpdateParams.ImageType = infraEnv.Spec.ImageType
+	}
```

Comment:
For the update, do you think it's a good idea to get the image type like how you did it in the [create function](https://github.com/openshift/assisted-service/pull/6998/files#diff-402c091b3839c89d4be13bcb028e7bfaf13c9380e01d430556b7e7b27d356f1eR503) and compare that to what's stored in the db? 
```suggestion
        imageType := r.getImageType(log, infraEnv)
	if imageType != common.ImageTypeValue(internalInfraEnv.Type) {
		updateParams.InfraEnvUpdateParams.ImageType = imageType
	}
```

Reply:
`getImageType` calls to `GetInfraEnvIsoImageType` that return default values on creating an infraenv

Reply:
Yes and that should either match the value stored in the DB or if it's different, then we should update it